curriculum/test/test-challenges.js

```diff
@@ -351,42 +347,43 @@ A required file can not have both a src and a link: src = ${src}, link = ${link}
   await runTestInJsdom(dom, test.testString);
 }
 
-async function evaluateJsTest({
-  solution,
-  files,
-  test
-}) {
-
+async function evaluateJsTest({ solution, files, test }) {
   const virtualConsole = new jsdom.VirtualConsole();
   const dom = new JSDOM('', { runScripts: 'dangerously', virtualConsole });
 
   const { head = '', contents = '', tail = '' } = files.js;
   let scriptString = '';
   if (!solution) {
     solution = contents;
-    scriptString = head + '\n' + contents + '\n' + tail + '\n';
     try {
-      // eslint-disable-next-line
-      new vm.Script(scriptString);
+      scriptString =
```

Comment:
Why would we not concat the string then run transform?

Reply:
1. We do the same in the client app
2. For some challenges concatenation before transform gives a wrong result. For example https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/understand-the-differences-between-import-and-require.english.md

If we concatenate before transform we'll get:
Before transform
```js
window.require = function (str) {
  if (str === 'string_functions') {
    return {
      capitalizeString: function capitalizeString(str) {
        return str.toUpperCase();
      }
    };
  }
};
'use strict';
import { capitalizeString } from 'string_functions';
capitalizeString('hello!');

```
After transform
```js
'use strict';

var _string_functions = require('string_functions');

window.require = function (str) {
  if (str === 'string_functions') {
    return {
      capitalizeString: function capitalizeString(str) {
        return str.toUpperCase();
      }
    };
  }
};

'use strict';

(0, _string_functions.capitalizeString)('hello!');
```
which gives an error `Cannot find module 'string_functions'`.